components/InputStates.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * InputStates.js
+ * 
+ * Manages the input states for a component's ports
+ */
+
+var _ = require('underscore');
+
+var inputStates = [];
```

Comment:
Yes, this is singleton for this module instance.  If you want one per VNI, then you'll need to create one object per VNI.   Will see about that in the next pass of this.  But there is no VNI in this check in, you will note.